### PR TITLE
DM-33661: narrativelog: initial chart

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

**Review by athornton — DISMISSED** [review]
Same as for the phalanx PR, ensure that the settings for black/flake8 match what we're doing elsewhere.  Otherwise looks good.  I don't think we've done horizontal pod autoscaling elsewhere, but no reason not to give it a try.

I think we probably want to move the .helmignores up and out of the individual charts, but that's another PR for another time.

File: charts/narrativelog/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v2
+name: narrativelog
+description: Exposure log service
```
Comment (jonathansick) [inline charts/narrativelog/Chart.yaml:3]: Update for narrative log?

**Review by jonathansick — CHANGES_REQUESTED** [review]
If possible, I'd recommend splitting this PR into two: one for adding narrativelog and one for modifying the chart repo's testing infrastructure.

Was this chart made made with our recommended template (https://github.com/lsst-sqre/charts#adding-a-new-chart-to-this-repository)? It seems like some of the podSecurityContext and securityContext defaults in values aren't being set as well as duplication between imagePullSecrets and pull_secret.

**Comment by r-owen** [comment]
I moved the linting changes to another ticket. All that's left here should be the narrativelog files.

File: charts/narrativelog/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
```
Comment (jonathansick) [inline charts/narrativelog/templates/deployment.yaml:37]: Also we've started to bake in a securityContext for our containers: https://github.com/lsst-sqre/charts/blob/b37a33f1f0298a0e1e16bf18757655eafbfb0e99/rsp-starter/templates/deployment.yaml#L34-L39

File: charts/narrativelog/templates/ingress.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+{{- if .Values.ingress.enabled -}}
```
Comment (jonathansick) [inline charts/narrativelog/templates/ingress.yaml:1]: A more standardized ingress that works well with the Phalanx/RSP auth is https://github.com/lsst-sqre/charts/blob/master/rsp-starter/templates/ingress.yaml

**Review by jonathansick — CHANGES_REQUESTED** [review]
I've got some line comments on areas where we want to move to our standardized chart regarding security contexts and ingress. Another thing missing is a NetworkPolicy, which you can get from https://github.com/lsst-sqre/charts/blob/master/rsp-starter/templates/networkpolicy.yaml

File: charts/narrativelog/templates/ingress.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  {{- if .Values.ingress.className }}
+  ingressClassName: {{ .Values.ingress.className }}
+  {{- end }}
+  rules:
+    - host: {{ required "ingress.host must be set" .Values.ingress.host | quote }}
+      http:
+        paths:
+          - path: {{ .Values.ingress.path }}
+            pathType: {{ default "Prefix" .Values.ingress.pathType }}
+            backend:
```
Comment (jonathansick) [inline charts/narrativelog/templates/ingress.yaml:30]: Try an extra `{{- end }}` above backend?

See https://github.com/lsst-sqre/charts/blob/b37a33f1f0298a0e1e16bf18757655eafbfb0e99/rsp-starter/templates/ingress.yaml#L30

**Comment by r-owen** [comment]
All changes made, after fixing an error in the provided ingress file (an extra "end" before "backend").

**Review by jonathansick — APPROVED** [review]
Ok, this looks good. there's there the hold-over `pull_secret` the values.yaml but it's not used.

**Comment by r-owen** [comment]
Thanks. I removed the extra pull_secret from values.yaml. Now I just need @athornton  to sign off on it.

**Pull request merged**